cdap-client/src/main/java/co/cask/cdap/client/ProgramClient.java

```diff
@@ -353,7 +355,18 @@ public void stopAll(NamespaceId namespace)
           ProgramId program = appId.program(programRecord.getType(), programRecord.getName());
           String status = this.getStatus(program);
           if (!status.equals("STOPPED")) {
-            this.stop(program);
+            try {
+              this.stop(program);
+            } catch (IOException ioe) {
+              // ProgramClient#stop calls RestClient, which throws an IOException if the HTTP response code is 400,
```

Comment:
Seems like we should fix the library to not to throw IOException. I see that we already handled the 404 case.

https://github.com/caskdata/common/blob/develop/common-http/src/main/java/co/cask/common/http/HttpRequests.java#L121